ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -206,54 +229,63 @@ class ContractsService(
   )(implicit
       lc: LoggingContextOf[InstanceUUID]
   ): SearchResult[Error \/ domain.ActiveContract[LfValue]] =
-    retrieveAll(jwt, jwtPayload.parties)
+    retrieveAll(jwt, toLedgerId(jwtPayload.ledgerId), jwtPayload.parties)
 
   def retrieveAll(
       jwt: Jwt,
+      ledgerId: LedgerApiDomain.LedgerId,
       parties: OneAnd[Set, domain.Party],
   )(implicit
       lc: LoggingContextOf[InstanceUUID]
   ): SearchResult[Error \/ domain.ActiveContract[LfValue]] =
     domain.OkResponse(
-      Source(allTemplateIds()).flatMapConcat(x => searchInMemoryOneTpId(jwt, parties, x, _ => true))
+      Source(allTemplateIds()).flatMapConcat(x =>
+        searchInMemoryOneTpId(jwt, ledgerId, parties, x, _ => true)
+      )
     )
 
   def search(
       jwt: Jwt,
       jwtPayload: JwtPayload,
       request: GetActiveContractsRequest,
   )(implicit
-      lc: LoggingContextOf[InstanceUUID]
-  ): SearchResult[Error \/ domain.ActiveContract[JsValue]] =
-    search(jwt, jwtPayload.parties, request.templateIds, request.query)
+      lc: LoggingContextOf[InstanceUUID with RequestID]
+  ): Future[SearchResult[Error \/ domain.ActiveContract[JsValue]]] =
+    search(
+      jwt,
+      toLedgerId(jwtPayload.ledgerId),
+      jwtPayload.parties,
+      request.templateIds,
+      request.query,
+    )
 
   def search(
       jwt: Jwt,
+      ledgerId: LedgerApiDomain.LedgerId,
       parties: OneAnd[Set, domain.Party],
       templateIds: OneAnd[Set, domain.TemplateId.OptionalPkg],
       queryParams: Map[String, JsValue],
   )(implicit
-      lc: LoggingContextOf[InstanceUUID]
-  ): SearchResult[Error \/ domain.ActiveContract[JsValue]] = {
-
-    val (resolvedTemplateIds, unresolvedTemplateIds) = resolveTemplateIds(templateIds)
+      lc: LoggingContextOf[InstanceUUID with RequestID]
+  ): Future[SearchResult[Error \/ domain.ActiveContract[JsValue]]] = for {
+    res <- resolveTemplateIds(jwt, ledgerId)(templateIds)
+    (resolvedTemplateIds, unresolvedTemplateIds) = res
 
-    val warnings: Option[domain.UnknownTemplateIds] =
+    warnings: Option[domain.UnknownTemplateIds] =
```

Comment:
Congrats, you taught me a Scala thing. I didn't know ascription was allowed here.

Reply:
![You taught me a Scala thing](https://user-images.githubusercontent.com/43749967/131804116-ee38a300-59bb-49bd-a5ec-2b62753182a1.png)